os/fs/smartfs/smartfs_smart.c

```diff
@@ -289,21 +288,43 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 			}
 
 			/* First try to find an invalid or empty entry available in one of the chained parent sectors */
-			ret = smartfs_find_availableentry(fs, &parentdirsector, &newentry);
+			ret = smartfs_find_availableentry(fs, &sf->entry);
 			if (ret != OK) {
 				/* find_availableentry encountered a problem with lower layer operations, return error */
 				fdbg("smartfs_find_availableentry() encountered a problem, unable to find entry for writing\n");
 				goto errout_with_buffer;
 			}
 
 			/* At this point, either an available entry was found or a new one has been created */
-			newentry.firstsector = data_sector;
+			sf->entry.firstsector = data_sector;
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+			/* If CRC is enabled, hold the entry for writing later */
+			/* Mark flags to indicate that the entry will be written to a new sector in the parent directory.
+			   This sector will then be chained in the end.
+			 */
+			sf->bflags |= SMARTFS_BFLAG_NEW_ENTRY;
+			/* Set sf->entry flags (to be writtne later) and store them */
+
+			sf->entry.flags = SMARTFS_ERASEDSTATE_16BIT;
+			smartfs_set_entry_flags(&sf->entry, SMARTFS_DIRENT_TYPE_FILE, mode);
 
-			ret = smartfs_writeentry(fs, newentry, filename, SMARTFS_DIRENT_TYPE_FILE, mode, &sf->entry, parentdirsector);
+			if (sf->entry.name == NULL) {
+				sf->entry.name = (FAR char *)kmm_malloc(fs->fs_llformat.namesize + 1);
+				if (sf->entry.name == NULL) {
+					ret = -ENOMEM;
+					goto errout_with_buffer;
+				}
+			}
+			memset(sf->entry.name, 0, fs->fs_llformat.namesize + 1);
```

Comment:
Please use zalloc in all places instead of memset zero.

Reply:
Currently in line with all other entry->name space allocations.
All of which are to be changed an upcoming next commit.